File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Session/PostSessionDto.java
package com.salesianos.triana.BeElite.dto.Session;

import com.salesianos.triana.BeElite.dto.Block.BlockDto;
import com.salesianos.triana.BeElite.dto.Block.PostBlockDto;
import com.salesianos.triana.BeElite.model.Composite_Ids.SessionId;
import com.salesianos.triana.BeElite.model.Composite_Ids.WeekId;
import com.salesianos.triana.BeElite.model.Session;
import com.salesianos.triana.BeElite.model.Week;
import com.salesianos.triana.BeElite.service.BlockService;
import com.salesianos.triana.BeElite.service.SessionService;
import lombok.Builder;

import java.time.LocalDate;
import java.util.List;

@Builder
public record PostSessionDto(
        Long session_number,
        LocalDate date,
        String title,
        String subtitle,
        List<PostBlockDto> blocks,
        Integer same_day_session_number
) {

    public static PostSessionDto of(Session s){
        return PostSessionDto.builder()
                .session_number(s.getId().getSession_number())
                .date(s.getDate())
                .title(s.getTitle())
                .subtitle(s.getSubtitle())
                .blocks(s.getBlocks() != null ?
                        s.getBlocks().stream().map(PostBlockDto::of).toList() :
                        List.of())
                .same_day_session_number(s.getSame_day_session_number())
                .build();
    }
}

File: frontend/be_elite/lib/models/Session/post_session_dto/post_session_dto.dart
import 'post_block_dto.dart';

class PostSessionDto {
	int? sessionNumber;
	String? date;
	String? title;
	String? subtitle;
	List<PostBlockDto>? blocks;
	int? sameDaySessionNumber;

	PostSessionDto({
		this.sessionNumber, 
		this.date, 
		this.title, 
		this.subtitle, 
		this.blocks, 
		this.sameDaySessionNumber, 
	});

	factory PostSessionDto.fromJson(Map<String, dynamic> json) {
		return PostSessionDto(
			sessionNumber: json['session_number'] as int?,
			date: json['date'] as String?,
			title: json['title'] as String?,
			subtitle: json['subtitle'] as String?,
			blocks: (json['blocks'] as List<dynamic>?)
						?.map((e) => PostBlockDto.fromJson(e as Map<String, dynamic>))
						.toList(),
			sameDaySessionNumber: json['same_day_session_number'] as int?,
		);
	}



	Map<String, dynamic> toJson() => {
				'session_number': sessionNumber,
				'date': date,
				'title': title,
				'subtitle': subtitle,
				'blocks': blocks?.map((e) => e.toJson()).toList(),
				'same_day_session_number': sameDaySessionNumber,
			};
}
